### PR TITLE
Enable agent registration on remote sites

### PR DESCRIPTION
**Pull request by robin-tribe29**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

#219 

## What is the new behavior?

- Enable agent registration on remote sites
- Download is not possible yet

### CODE REVIEW

File: roles/agent/README.md [hunk 1]
```diff
@@ -46,6 +46,14 @@ The port of the web interface of your Checkmk server. Defaults to port 80 for ht
 
 The name of your Checkmk site.
 
+    checkmk_agent_registration_server: "{{ checkmk_agent_server }}"
+
+The server you want to use for registration tasks (Agent updates and TLS encryption). Defaults to {{ checkmk_agent_server }}.
+
+    checkmk_agent_registration_site: "{{ checkmk_agent_site }}"
+
+The server you want to use for registration tasks (Agent updates and TLS encryption). Defaults to {{ checkmk_agent_site }}.
```
Comment (lgetwan) [inline roles/agent/README.md:55]: I guess, you meant "site".

File: roles/agent/tasks/main.yml [hunk 2]
```diff
@@ -47,7 +50,7 @@
 
 - name: "Create host on server."
   tribe29.checkmk.host:
-    server_url: "{{ checkmk_agent_protocol }}://{{ checkmk_agent_server }}:{{ checkmk_agent_port }}/"
+    server_url: "{{ checkmk_agent_protocol }}://{{ checkmk_agent_registration_server }}:{{ checkmk_agent_port }}/"
```
Comment (lgetwan) [inline roles/agent/tasks/main.yml:53]: Shouldn't the host always be created on the central site? And the API to talk to should also be the central one, shouldn't it?

Comment (robin-tribe29) [reply roles/agent/tasks/main.yml:53]: Interesting. I am almost certain, that this worked in my tests. I think I will need to reiterate.
Can you test this on your end too? Just to have a separate view.

File: roles/agent/README.md (hunk 1 above)
Comment (robin-tribe29) [reply roles/agent/README.md:55]: Jup, copa-paste error. Already fixed. :v:

File: roles/agent/tasks/main.yml (hunk 2 above)
Comment (lgetwan) [reply roles/agent/tasks/main.yml:53]: It's even complaining that WATO is disabled on the remote site if I use checkmk_agent_registration_server, here.
To configure where a host should reside, the host attributes should contain the site. E.g.
`checkmk_agent_host_attributes:  
    site: "multi_remote_1"`

File: roles/agent/tasks/main.yml [hunk 3]
```diff
@@ -78,7 +81,7 @@
   become: true
   ansible.builtin.shell: |
     cmk-update-agent register -H {{ checkmk_agent_host_name }} \
-    -s {{ checkmk_agent_server }} -i {{ checkmk_agent_site }} -p {{ checkmk_agent_protocol }} \
+    -s {{ checkmk_agent_registration_server }} -i {{ checkmk_agent_site }} -p {{ checkmk_agent_protocol }} \
```
Comment (lgetwan) [inline roles/agent/tasks/main.yml:84]: Should be 
`{{ checkmk_agent_registration_site }}`

File: roles/agent/tasks/main.yml [hunk 4]
```diff
@@ -94,7 +97,7 @@
...
```
Comment (lgetwan) [inline roles/agent/tasks/main.yml:100]: Should be 
`{{ checkmk_agent_registration_site }}`

**Review by lgetwan — COMMENTED** [review]
We also need an "Activate Changes" after creating the host. Otherwise, the TLS-Registration is complaining "Host xyz does not exist."

File: roles/agent/tasks/main.yml (hunk 2 above)
Comment (DominikBauer1) [reply roles/agent/tasks/main.yml:53]: I can confirm this. If WATO is enabled it creates an inconsistency where the host is available on the remote site but not on central.

**Comment by DominikBauer1** [comment]
solves #232

File: roles/agent/tasks/main.yml (hunk 2 above)
Comment (robin-tribe29) [reply roles/agent/tasks/main.yml]: Fixes are pushed, along with automatic change activation.
Please re-test and do let me know. :V:

**Pull request merged**